Projects/private-ca/roles/vault/tasks/main.yml

```diff
@@ -62,6 +64,11 @@
   shell: vault write -client-key /etc/ssl/cert-key.pem -client-cert /etc/ssl/cert.pem /auth/token/roles/nomad-cluster @/etc/vault/nomad-cluster-role.json
   run_once: true
 
-#- name: Create nomad-server policy
-#  shell: vault operator init -client-key /etc/ssl/cert-key.pem -client-cert /etc/ssl/cert.pem
-#  run_once: true
\ No newline at end of file
+- name: Create the Nomad server token
+  shell: vault token create -client-key /etc/ssl/cert-key.pem -client-cert /etc/ssl/cert.pem -policy nomad-server -period 72h -orphan
+  register: nomad_server_token
+
+- name: Write the server token to tmpfs
+  copy:
+    content: "VAULT_TOKEN={{ nomad_server_token.stdout_lines[2][19:] }}"
+    dest: /run/nomad-vault-token
```

Comment:
As soon as the initialization, the role and the policy set up, we are now ready to generate the Vault token for Nomad to use. This way we don't end up with the security problem of having a root token being used by everything everywhere, with too much access.